### PR TITLE
RSSACR

### PR DESCRIPTION
**Pull request by Vilin97**
Wrote the RSSA-CR method. It passes the current tests, and so far performs a bit worse than RSSA.

### CODE REVIEW

File: test/trajectory_test.jl [hunk 1]
```diff
@@ -0,0 +1,101 @@
+using DiffEqBase, DiffEqJump
```
Comment (isaacsas) [inline test/trajectory_test.jl]: Is this test adding something over the other tests? If so, it should be added to runtests too, if not remove it.

File: test/geneexpr_test.jl [hunk 2]
```diff
@@ -102,13 +102,13 @@ if dotestmean
...
-#     # exact methods
-#     for alg in SSAalgs
-#         println("Solving with method: ", typeof(alg), ", using SSAStepper")
-#         jump_prob = JumpProblem(prob, alg, majumps, vartojumps_map=spec_to_dep_jumps, jumptovars_map=jump_to_dep_specs)
-#         @btime solve($jump_prob, SSAStepper())
-#     end
-#     println()
-# end
+# benchmark performance
+if dobenchmark
```
Comment (isaacsas) [inline test/geneexpr_test.jl:106]: Leave these commented out so we don't need the BenchmarkTools dependency.

File: test/geneexpr_test.jl [hunk 3]
```diff
@@ -3,14 +3,14 @@ using Test, Statistics
 
 # using Plots; plotlyjs()
 doplot = false
-# using BenchmarkTools
-# dobenchmark = false
+using BenchmarkTools
```
Comment (isaacsas) [inline test/geneexpr_test.jl:6]: comment out

File: test/extinction_test.jl [hunk 4]
```diff
@@ -22,6 +19,9 @@ dprob = DiscreteProblem(u0,(0.,100.),rates)
 
 algs = DiffEqJump.JUMP_AGGREGATORS
 
+u0 = [10]
+dprob = DiscreteProblem(u0,(0.,10.),rates)
```
Comment (isaacsas) [inline test/extinction_test.jl:23]: Why did you decrease this to 10? We were getting occasional test failures at 10 before (rare events where it hadn't gone extinct by that time).

**Comment by ChrisRackauckas** [comment]
This is looking great!

**Comment by isaacsas** [comment]
@Vilin97 Is this ok to merge? We can work on optimizing the speed on the BCR benchmark as a follow up, but it looks ok accuracy wise now.

**Comment by Vilin97** [comment]
Yes, okay to merge

**Comment by isaacsas** [comment]
Nice first PR!

**Pull request merged**